exporter/exporterhelper/internal/sizer/metrics_sizer.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sizer // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal/sizer"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pmetric"
+) // MetricsCountSizer returns the nunmber of metrics entries.
+
+type MetricsSizer interface {
+	MetricsSize(md pmetric.Metrics) (count int)
+	ResourceMetricsSize(rm pmetric.ResourceMetrics) (count int)
+	ScopeMetricsSize(sm pmetric.ScopeMetrics) (count int)
+	MetricSize(m pmetric.Metric) int
+	DeltaSize(newItemSize int) int
+	NumberDataPointSize(ndp pmetric.NumberDataPoint) int
+	HistogramDataPointSize(hdp pmetric.HistogramDataPoint) int
+	ExponentialHistogramDataPointSize(ehdp pmetric.ExponentialHistogramDataPoint) int
+	SummaryDataPointSize(sdps pmetric.SummaryDataPoint) int
+}
+
+type MetricsBytesSizer struct {
+	pmetric.ProtoMarshaler
+}
+
+var _ MetricsSizer = &MetricsBytesSizer{}
+
+// DeltaSize() returns the delta size of a proto slice when a new item is added.
+// Example:
+//
+//	prevSize := proto1.Size()
+//	proto1.RepeatedField().AppendEmpty() = proto2
+//
+// Then currSize of proto1 can be calculated as
+//
+//	currSize := (prevSize + sizer.DeltaSize(proto2.Size()))
+//
+// This is derived from opentelemetry-collector/pdata/internal/data/protogen/metrics/v1/metrics.pb.go
+// which is generated with gogo/protobuf.
+func (s *MetricsBytesSizer) DeltaSize(newItemSize int) int {
+	return 1 + newItemSize + sov(uint64(newItemSize)) //nolint:gosec // disable G115
+}
```

Comment:
I would extract this into a struct, `proto_delta_sizer.go`, and embed protoDeltaSizer into the 
MetricsBytesSizer.

```
type protoDeltaSizer struct{}

func (s *protoDeltaSizer) DeltaSize(newItemSize int) int {
	return 1 + newItemSize + sov(uint64(newItemSize)) //nolint:gosec // disable G115
}
```

Move tests as well, then we remove duplicate code.